src/odemis/driver/ephemeron.py

```diff
@@ -0,0 +1,788 @@
+        """
+        data_var = await self.server.nodes.objects.add_variable(
+            self.idx,
+            name,
+            bytes(data.tolist()),
+        )
+        await data_var.set_writable()
+
+
+class EBICDataFlow(model.DataFlow):
+    def __init__(self, detector):
+        """
+        detector (semcomedi.Detector): the detector that the dataflow corresponds to
+        sem (semcomedi.SEMComedi): the SEM
+        """
+        model.DataFlow.__init__(self)
+        self.component = weakref.ref(detector)
+
+        self._sync_event = None  # event to be synchronised on, or None
+        self._evtq = None  # a Queue to store received events (= float, time of the event)
+
+    # start/stop_generate are _never_ called simultaneously (thread-safe)
+    def start_generate(self):
+        try:
+            self.component().start_acquire(self.notify)
+        except ReferenceError:
+            # sem/component has been deleted, it's all fine, we'll be GC'd soon
+            pass
+
+    def stop_generate(self):
+        try:
+            self.component().stop_acquire()
+            # Note that after that acquisition might still go on for a short time
+        except ReferenceError:
+            # sem/component has been deleted, it's all fine, we'll be GC'd soon
+            pass
+
+    def synchronizedOn(self, event):
+        """
+        Synchronize the acquisition on the given event. Every time the event is
+        triggered, the scanner will start a new acquisition/scan.
+        The DataFlow can be synchronized only with one Event at a time.
+        However, each DataFlow can be synchronized separately. The scan will
+        only start once each active DataFlow has received an event.
+        event (model.Event or None): event to synchronize with. Use None to disable synchronization.
+        """
+        super().synchronizedOn(event)
+        if self._sync_event == event:
+            return
+
+        if self._sync_event:
+            self._sync_event.unsubscribe(self)
+            if not event:
+                self._evtq.put(None)  # in case it was waiting for this event
+
+        self._sync_event = event
+        if self._sync_event:
+            # if the df is synchronized, the subscribers probably don't want to skip some data
+            self._evtq = queue.Queue()  # to be sure it's empty
+            self._sync_event.subscribe(self)
+
+    @oneway
+    def onEvent(self):
+        """
+        Called by the Event when it is triggered
+        """
+        if not self._evtq.empty():
+            logging.warning("Received synchronization event but already %d queued",
+                            self._evtq.qsize())
+
+        self._evtq.put(time.time())
+
+    def _waitSync(self):
+        """
+        Block until the Event on which the dataflow is synchronised has been
+          received. If the DataFlow is not synchronised on any event, this
+          method immediately returns
+        """
+        if self._sync_event:
```

Comment:
I don't think you should try to handle software triggers (yet). => Just drop that code.
Possibly, you could still leave the `synchronizedOn()` function, to detect the user requested a hardware trigger (see andorcam2), and if that's not the case, raise an error at acquisition complaining that the device only works if synchronized (but I think it just makes our life harder when debugging... so I would not even bother).